### ISSUE TITLE
when the `state` is `overridden` in the `dcnm_interface` module, default policy of `leaf` should be trunk

### ISSUE THREAD

**Issue by dsx1123** [CiscoDevNet/ansible-dcnm#282]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### NDFC version

* V 12.1.3e

### Affected module(s)

* dcnm_interface

### Ansible Playbook

```yaml
# Copy-paste your anisble playbook here 
---
- hosts: ndfc2
  gather_facts: false
  vars:
    fabric: "{{ fabric2 }}"
  tasks:
    - name: Test interface overridden
      cisco.dcnm.dcnm_interface:
        fabric: "{{ fabric.name }}"
        state: overridden
        config: []
```

### Expected Behavior

The ethernet interfaces of leaf switches should be reset to the trunk interface, but ethernet interfaces on the rest of the roles should be set to routed interfaces

**Comment by mmudigon** [comment]
@dsx1123 Hi Shangxin,

Will look into the issue and fix it

**Comment by praveenramoorthy** [comment]
Fixed as part of #287
